remoteconfig_test.go

```diff
@@ -591,52 +591,30 @@ func (s *RemoteConfigSuite) TestLoadConfigFromURLError() {
 }
 
 func (s *RemoteConfigSuite) TestReadJSONValidate() {
-	configJSON := `
-	{
-		"sqs_client" : {
-			"region" : "us-east-1",
-			"endpoint" : "http://localhost:3000/sqs"
-		},
-		"sqs_queue" : {
-			"region" : "us-east-1",
-			"aws_account_id" : "345833302425",
-			"queue_name" : "testQueue"
-		},
-		"dynamodb_client" : {
-			"region" : "us-east-1",
-			"endpoint" : "http://localhost:8000/dynamodb"
-		},
-		"dynamodb_table" : {
-			"table_name" : "testTable"
-		},
-		"str" : "testStr",
-		"storage_config" : {
-			"provider" : "aws",
-			"location" : "us-west-2"
-		},
-		"storage_config_slice" : [{
-			"provider" : "aws",
-			"location" : "us-west-2"
-		},
-		{
-			"provider" : "aws",
-			"location" : "us-east-1"
-		}],
-		"storage_config_map": {
-			"one": {
-				"provider": "aws",
-				"location": "us-west-2"
-			}
-		},
-		"str_slice": [ "hello" ],
-		"map_str_str": { "key": "value" }
-	}`
+	cfgBuffer := bytes.NewBufferString(validConfigJSON)
 
-	cfgBuffer := bytes.NewBufferString(configJSON)
+	c := &SampleConfig{}
+	err := ReadJSONValidate(cfgBuffer, c)
+	assert.Nil(s.T(), err)
+}
+
+func (s *RemoteConfigSuite) TestReadJSONParseEmbeddedStruct() {
+	cfgBuffer := bytes.NewBufferString(validConfigJSON)
 
 	c := &SampleConfig{}
 	err := ReadJSONValidate(cfgBuffer, c)
 	assert.Nil(s.T(), err)
+	assert.NotNil(s.T(), c.EmbeddedStr)
+	assert.EqualValues(s.T(), "abc", *c.EmbeddedStr)
+}
```

Comment:
could check the embeddedint, but really i guess if that's broken, it's really more a set fire to everything situation.